### PR TITLE
Clarify the behaviour of Pattern when used with methods like str::contains

### LINKED ISSUE #45507: std::str::pattern::Pattern documentation
Correct me if I'm wrong, but I don't think it's possible to find out from the currently available online documentation if the following code returns true or false:
```rust
"ab".contains(&['b', 'a'][..])
```
Is searching for a slice of chars the same as searching for the string of the contatenation? Or is this a set match?

There is no information on this in the following places:
* https://doc.rust-lang.org/std/primitive.str.html#method.contains
* https://doc.rust-lang.org/std/str/pattern/trait.Pattern.html
* https://doc.rust-lang.org/std/str/pattern/struct.CharSliceSearcher.html

### PR DESCRIPTION
**Pull request by poliorcetics**
Fixes #45507.

I used the previous work by @Emerentius (thanks !), added a paragraph and checked the links (they work for me but I'm not against someone else checking them too).

### CODE REVIEW

**Comment by poliorcetics** [comment]
@hanna-kruppe I'm done with this, if you have the time could you review it ?

**Comment by hanna-kruppe** [comment]
Apologies, I was extremely busy until yesterday. I'll take a look soon.

**Comment by poliorcetics** [comment]
Don't worry, it's not urgent at all ! That was not intended to pressure you

**Comment by hanna-kruppe** [comment]
For the record, this information is also documented on the `Pattern` *impls* (nowadays; perhaps not when https://github.com/rust-lang/rust/issues/45507 was filed), but I think duplicating this information in a central and more obvious location is helpful.

The content LGTM, CI will tell us if the links work. @bors r+

File: src/libcore/str/pattern.rs [hunk 1]
```diff
@@ -60,6 +60,43 @@ use crate::slice::memchr;
...
+///
+/// Depending on the type of the pattern, the behaviour of methods like
+/// [`str::find`] and [`str::contains`] can change. The table below describes
+/// some of those behaviours.
+///
+/// | Pattern type             | Match condition                           |
+/// |--------------------------|-------------------------------------------|
+/// | `&str`                   | is substring                              |
+/// | `char`                   | is contained in string                    |
+/// | `&[char]                 | any char in slice is contained in string  |
```
Comment (lzutao) [inline src/libcore/str/pattern.rs:72]: Missed the ending "`".

Comment (lzutao) [reply src/libcore/str/pattern.rs:72]: Resolving in #73336 .

**Pull request merged**